### PR TITLE
Provides a solution for running ut with docker and a solution for running locally

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
Provides a solution for running ut with docker and a solution for running locally
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by LiZhenCheng9527** [comment]
Provides a script to run go unit test in docker and a script to run go unit test locally.
There will be documentation to go along with this that explains the scenarios in which the two scripts are used.

File: hack/run-ut.sh [hunk 1]
```diff
@@ -2,6 +2,51 @@
...
+
+function docker_run_go_ut() {
+    local container_id=$1
+    docker exec $container_id go test -v -vet=off -coverprofile=coverage.out ./pkg/...
+}
+
+function run_go_ut_local() {
+    bash $ROOT_DIR/build.sh
+    export PKG_CONFIG_PATH=$ROOT_DIR/mk
+    go test -v -vet=off -coverprofile=coverage.out ./pkg/...
```
Comment (hzxuzhonghu) [inline hack/run-ut.sh:15]: we should not generate coverage by default

File: Makefile [hunk 2]
```diff
@@ -151,7 +151,8 @@ docker: build
 format:
 	./hack/format.sh
 
-test: build
+.PHONY: test
+test:
 	./hack/run-ut.sh
```
Comment (hzxuzhonghu) [inline Makefile:156]: how can we pass param to this script

File: hack/run-ut.sh [hunk 3]
```diff
@@ -2,6 +2,51 @@
...
+}
+
+# Running go ut with docker by default
+if [ -z "$1" -o "$1" == "-d"  -o  "$1" == "--docker" ]; then
+    run_go_ut_in_docker
+    exit
+fi
+
+if [ "$1" == "-l"  -o  "$1" == "--local" ]; then
+    run_go_ut_local
```
Comment (hzxuzhonghu) [inline hack/run-ut.sh:38]: I met an error running with `./hack/run-ut.sh -l`

Comment (LiZhenCheng9527) [reply hack/run-ut.sh:38]: What is the error?

File: Makefile (hunk 2 above)
Comment (LiZhenCheng9527) [reply Makefile]: You can now pass param by specifying UT_TYPE.

```sh
make test UT_TYPE=--docker
make test UT_TYPE=--local
```

File: Makefile [hunk 4]
```diff
@@ -17,6 +17,7 @@ GIT_COMMIT_HASH ?= $(shell git rev-parse HEAD)
 GIT_TREESTATE=$(shell if [ -n "$(git status --porcelain)" ]; then echo "dirty"; else echo "clean"; fi)
 BUILD_DATE = $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
 ROOT_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+UT_TYPE = --docker
```
Comment (YaoZengzeng) [inline Makefile:20]:
```suggestion
UT_ENV= --docker
```

maybe better, means different UT running environment

Comment (hzxuzhonghu) [reply Makefile:20]: How about  `make ut RUN_WITH_CONTAINER=0` or `make ut RUN_WITH_CONTAINER=1`

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**